File: backend/models/mistake.py
from pydantic import BaseModel, Field
from typing import Optional, Literal
from datetime import datetime
from uuid import UUID

MasteryLevel = Literal["low", "medium", "critical"]
Category = Literal["physical", "human", "regional"]

class MistakeBase(BaseModel):
    """Base mistake schema."""
    title: str = Field(..., min_length=1, max_length=200)
    question: Optional[str] = None
    category: Category = "physical"
    mastery_level: MasteryLevel = "low"
    image_url: Optional[str] = None

class MistakeCreate(MistakeBase):
    """Schema for creating a mistake entry."""
    user_id: UUID

class MistakeUpdate(BaseModel):
    """Schema for updating a mistake entry."""
    mastery_level: Optional[MasteryLevel] = None
    question: Optional[str] = None

class Mistake(MistakeBase):
    """Complete mistake schema."""
    id: UUID
    user_id: UUID
    added_at: datetime
    
    class Config:
        from_attributes = True

File: bankend-springboot/bankend-common/bankend-common-frame/src/main/java/com/flittly/bankendspringboot/entity/Mistake.java
package com.flittly.bankendspringboot.entity;

import com.flittly.bankendspringboot.entity.enums.CategoryType;
import com.flittly.bankendspringboot.entity.enums.MasteryLevel;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
public class Mistake {
    private UUID id;
    private UUID userId;
    private String title;
    private String question;
    private CategoryType category;
    private MasteryLevel masteryLevel;
    private String imageUrl;
    private String explanation;
    private LocalDateTime addedAt;
    private LocalDateTime updatedAt;
}
